src/implementations/truncation.jl

```diff
@@ -182,15 +193,38 @@ function findtruncated(values::AbstractVector, strategy::TruncationKeepFiltered)
 end
 
 function findtruncated(values::AbstractVector, strategy::TruncationKeepBelow)
+    if issorted(values; by=abs, rev=true)
+        return convert(Vector{Int}, findtruncated_sorted(values, strategy))
+    else
+        return findtruncated_unsorted(values, strategy)
+    end
+end
+function findtruncated_sorted(values::AbstractVector, strategy::TruncationKeepBelow)
     atol = max(strategy.atol, strategy.rtol * first(values))
     i = @something findfirst(≤(atol), values) length(values) + 1
     return i:length(values)
 end
```

Comment:
Given that the input is assumed sorted, this can become:
```julia
    atol = max(strategy.atol, strategy.rtol * first(values))
    i = searchsortedfirst(values, atol)
    return i:length(values)
```